connector-builder-server/src/main/openapi/openapi.yaml

```diff
@@ -0,0 +1,246 @@
+                      items:
+                        type: string
+                    logs:
+                      type: array
+                      description: The LOG AirbyteMessages coming from the read operation for this page
+                      items:
+                        type: string
+                    request:
+                      type: object
+                      description: The HTTP request sent to the source API for this page
+                      required:
+                        - url
+                        - headers
+                      properties:
+                        url:
+                          type: string
+                          description: URL that the request was sent to
+                        body:
+                          type: object
+                          description: The body of the HTTP request, if present
+                        headers:
+                          type: object
+                          description: The headers of the HTTP request
+                    response:
+                      type: object
+                      description: The response from the source API for this page
+                      required:
+                        - status
+                        - headers
+                      properties:
+                        status:
+                          type: integer
+                          description: The status of the response
+                        body:
+                          type: object
+                          description: The body of the HTTP response, if present
+                        headers:
+                          type: object
+                          description: The headers of the HTTP response
+    StreamReadRequestBody:
+      type: object
+      required:
+        - definition
+        - stream
+      properties:
+        definition:
+          $ref: "#/components/schemas/ConnectorDefinitionBody"
+          description: The config-based connector definition contents
+        stream:
+          type: string
+          description: Name of the stream to read
+        numSlices:
+          type: integer
+          description: Number of stream slices to read from the source
+          default: 1
+        numPages:
+          type: integer
+          description: Number of pages to read from the source
+          default: 1
+        state:
+          type: object
+          description: State blob to use for incremental streams
+    ConnectorDefinitionBody:
+      $ref: ../../../../airbyte-cdk/python/airbyte_cdk/sources/declarative/config_component_schema.json
+    StreamsListRequestBody:
+      type: object
+      required:
+        - definition
+      properties:
+        definition:
+          $ref: "#/components/schemas/ConnectorDefinitionBody"
+          description: The config-based connector definition contents
+    StreamsListRead:
```

Comment:
would it be valuable to also return the stream's slices so they can be displayed?

Reply:
What do you mean exactly by "return the stream's slices"? Are you suggesting that this should return all of the possible slices that are configured for a given stream, e.g. if they are using a list slicer? And this would be displayed somewhere in the testing panel?

Reply:
Discussed this over zoom - the idea here is to have this return a list of all the slices that will be read for this stream, so that they can be displayed somewhere to provide visual feedback on how far through the stream slicing the user has made it. Added a commented-out block here as this can be done in a future phase (will add a note to the tech spec as well)